### PR TITLE
Ports: Add Simon Tatham's Puzzle Collection

### PR DESCRIPTION
**Pull request by Cleverking2003**
This PR adds a port of Simon Tatham's Portable Puzzle Collection. The port is missing a few features (such as the customization menu), however, all games are completely playable.
image

### CODE REVIEW

File: Ports/stpuzzles/package.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env -S bash ../.port_include.sh
+port=SimonTathamsPuzzles
+useconfigure=true
+version=1.0
+workdir=stpuzzles-serenity
+configopts="-DCMAKE_TOOLCHAIN_FILE=$SERENITY_SOURCE_DIR/Toolchain/CMake/CMakeToolchain.txt"
+files="https://github.com/Cleverking2003/stpuzzles/archive/refs/tags/serenity-${version}.zip stpuzzles.zip b9fcacbbc1a542f85fd3e62095aab52f88b6ec26533479103b55135ef822de08"
```
Comment (linusg) [inline Ports/stpuzzles/package.sh:7]: Unless there's a very good reason for it, please don't use a personal fork for ports sources. Use either a upstream repository/tarball/whatever or a repository under the SerenityOS GitHub organization.

File: Ports/stpuzzles/package.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+auth_type=sha256
+
+configure() {
+    run cmake $configopts -DCMAKE_CXX_FLAGS="-std=c++2a -O2"
+}
+
+install() {
+    mkdir -p "${SERENITY_INSTALL_ROOT}/opt/stpuzzles"
+    cp -r stpuzzles-serenity/* "${SERENITY_INSTALL_ROOT}/opt/stpuzzles"
```
Comment (linusg) [inline Ports/stpuzzles/package.sh:16]:
```suggestion
    run_nocd mkdir -p "${SERENITY_INSTALL_ROOT}/opt/stpuzzles"
    run_nocd cp -r stpuzzles-serenity/* "${SERENITY_INSTALL_ROOT}/opt/stpuzzles" 
```

File: Ports/stpuzzles/package.sh (hunk 1 above)
Comment (Cleverking2003) [reply Ports/stpuzzles/package.sh:7]: @linusg The changes I made to the original are too big to be made using patches and I can't create repositories in SerenityOS organization. What should I do?

Comment (linusg) [reply Ports/stpuzzles/package.sh]: Would you be happy to [transfer](https://docs.github.com/en/github/administering-a-repository/managing-repository-settings/transferring-a-repository) the repo? If not, I suppose we could fork it...

Comment (Cleverking2003) [reply Ports/stpuzzles/package.sh]: > Would you be happy to [transfer](https://docs.github.com/en/github/administering-a-repository/managing-repository-settings/transferring-a-repository) the repo? If not, I suppose we could fork it...

That's alright, I'll transfer it.

Comment (Cleverking2003) [reply Ports/stpuzzles/package.sh]: @linusg It seems that I can't transfer the repo because I'm not a member of the organization.

**Pull request merged**